var/spack/repos/builtin/packages/genfit/package.py

```diff
@@ -26,6 +26,7 @@ class Genfit(CMakePackage):
     depends_on('root')
     depends_on('root@:6.16.00', when='@b496504a')
     depends_on('eigen')
+    depends_on('googletest')
```

Comment:
We have a `test` deptype (which is used in some instances where `googletest` is declared as a dependency) but that is generally when the tests are build conditionally - it looks like as it is this package always builds the tests, so in that sense `googletest` is minimally a build (and possibly link) dependency (build+link is the default). If you want to run the tests on the package after building it you should also make this a run dependency like:


```
depends_on('googletest', type=('build', 'link', 'run'))
```

Reply:
Is the `test` phase still part of the build phase (environment wise) or is that run in a different environment afterwards? If it is the former than `build` and `link` should be enough. At least it was in my case.

The problem here is that `genfit` tries to build the tests if it finds *any* version of `googletest` in the envrionment. In my case it picked up one that was compiled with the pre gcc5 ABI and failed at linktime.

Since `genfit` also doesn't use CTest, I am not sure the tests could be invoked via spack at the moment. This would require to string together the necessary commands in a dedicated `test` function, I suppose?

I have submitted a PR to the `genfit` package that should solve at least a few of these issues, but I am not sure if and when that will be accepted.

Reply:
In short, I think that what you added is good so I will approve and merge for now, but below are some considerations regarding whether you might also want to add the `run` deptype

In more detail:

> Is the test phase still part of the build phase (environment wise) or is that run in a different environment afterwards? 

Yes: the test phase runs with all the environment settings for the build. 

> If it is the former than build and link should be enough. At least it was in my case.

I also agree that for the build that the `build/link` deptypes are sufficient. My understanding though was that the build generates test executables which become part of the installation prefix (i.e. executables in `<prefix>/bin`); the correct `gtest` wouldn't be available after the build unless it was marked as a run dependency.

If I misunderstood and the build needs `googletest` but then discards the tests (doesn't actually install them), then you are definitely right that `build/link` (and not `run`) is the right way to go.

> The problem here is that genfit tries to build the tests if it finds any version of googletest in the envrionment. In my case it picked up one that was compiled with the pre gcc5 ABI and failed at linktime.

I think you're right that it should be a build or run dependency then, vs. a test dependency (test dependencies are only included if you tell Spack you want to run tests, and it sounds like even when you don't that this would fail without `googletest`).